src/main/scala/org/grapheco/lynx/dataframe/DefaultDataFrameOperator.scala

```diff
@@ -13,13 +13,17 @@ import org.opencypher.v9_0.expressions.Expression
  */
 class DefaultDataFrameOperator(expressionEvaluator: ExpressionEvaluator) extends DataFrameOperator {
 
+  private def prepareColumns(df: DataFrame, columns: Seq[(String, Option[String])]): (Map[String, LynxType], Map[String, Int], Seq[Int]) = {
+      val sourceSchema = df.schema.toMap
+      val columnNameIndex = df.columnsName.zipWithIndex.toMap
+      val usedIndices = columns.map(_._1).map(columnNameIndex)
+      (sourceSchema, columnNameIndex, usedIndices)
+    }
   override def select(df: DataFrame, columns: Seq[(String, Option[String])]): DataFrame = {
-    val sourceSchema: Map[String, LynxType] = df.schema.toMap
-    val columnNameIndex: Map[String, Int] = df.columnsName.zipWithIndex.toMap
-    val newSchema: Seq[(String, LynxType)] = columns.map(column => column._2.getOrElse(column._1) -> sourceSchema(column._1))
-    val usedIndex: Seq[Int] = columns.map(_._1).map(columnNameIndex)
+    val (sourceSchema, _, usedIndices) = prepareColumns(df, columns)
```

Comment:
This seems to be just extracting these few operations, and it's not that different.